src/InterpLib/TextRangePrinting.ml

```diff
@@ -0,0 +1,219 @@
+(** Pretty printing of text range from Position.t *)
+open Position
+
+(* ========================================================================= *)
+(**  Options for underlining specified region  *)
+type underline_options =
+  | NoUnderline
+  (** Disable underlining *)
+
+  | UnderlineBegining
+  (** Point to only beginning of region *)
+
+  | UnderlineIfOneLine
+  (** Underline whole region, but only if specifies only one line *)
+
+  | UnderlineAlways
+  (** Underline whole region *)
+
+type options = {
+  context : int;
+  (** how many lines before region is to be printed *)
+
+  underline : t -> underline_options;
+  (** function that depending on region will select underlining option *)
+
+  add_line_numbers : bool;
+  (** should line numbers be added *)
+}
+
+let default_options = {
+  context = 2;
+  underline = (fun _ -> UnderlineIfOneLine);
+  add_line_numbers = true;
+}
+
+(* ========================================================================= *)
+(**  Printing to terminal with colors  *)
+
+let keyword_color = "\027[1;34m"
+
+type printing_color =
+  | Red
+  | Teal
+  | Yellow
+  | Black
+  | Green
+  | Blue
+  | Magenta
+  | Cyan
+  | White
+  | Default
+
+let color_to_string = function
+  | Black  -> "\027[30m"
+  | Red    -> "\027[31m"
+  | Green  -> "\027[32m"
+  | Yellow -> "\027[33m"
+  | Blue   -> "\027[34m"
+  | Magenta -> "\027[35m"
+  | Teal   -> "\027[36m"
+  | Cyan   -> "\027[36m"
+  | White  -> "\027[37m"
+  | Default -> "\027[0m"
+
+let bold_code_string = "\027[1m"
+let dim_code_string = "\027[2m"
+let italic_code_string = "\027[3m"
+let underline_code_string = "\027[4m"
+let blink_code_string = "\027[5m"
+let rapid_blink_code_string = "\027[6m"
+let reverse_code_string = "\027[7m"
+let hidden_code_string = "\027[8m"
+
+let reset_string = color_to_string Default
+
+let color_string color s =
+  if not !DblConfig.display_colors
+  then s
+  else color_to_string color ^ s ^ reset_string
+
+let bolden_string s =
+  if not !DblConfig.display_colors
+  then s
+  else bold_code_string ^ s ^ reset_string
+
+let underline_string s =
+  if not !DblConfig.display_colors
+  then s
+  else underline_code_string ^ s ^ reset_string
+
+let color_keywords line =
+  if not !DblConfig.display_colors then line else
+  let keywords =
+    [ "abstr"; "as"; "data"; "effect"
+    ; "effrow"; "else"; "end"; "extern"
+    ; "finally"; "fn"; "handle"; "handler"
+    ; "if"; "implicit"; "import"; "in"
+    ; "label"; "let"; "match"; "method"
+    ; "module"; "of"; "open"; "pub"
+    ; "rec"; "return"; "then"; "type"
+    ; "with"; "_"; "}"; "{"
+    ; "->"; "::"; "&&"; "||"; ";;"
+    ; "\\["; "\\]"; "=>"; "="; "|"
+    ] |> String.concat "\\|" in
```

Comment:
This code seems to be hard to maintain. Can we extract keywords from the lexer?
I see that there would be a small technical difficulty related to cyclic dependency of modules.

Reply:
I don't see a way around the dependency cycle, however I don't know that it will be so hard to maintain. I don't suppose we will be adding keywords that often and so many, so minor change once in a while won't be a problem.
When this list becomes outdated consequences will be apparent and harmless at the same time, also extremely easy to fix.

At the same time this isn't that necessary, there won't be that much loss in removing this part.

Reply:
I'm leaving this decision to you.

Reply:
I decided to remove this part. There are at least three reasons of this decision:

1. This code would be hard to maintain
2. It's far from being perfect: e.g., it colors keywords inside string literals
3. It provides functionality that is nice, but not necessary